pkg/k8s/watchers/watcher.go

```diff
@@ -410,6 +412,20 @@ func (k *K8sWatcher) EnableK8sWatcher(ctx context.Context, resources []string) e
 		return fmt.Errorf("error creating service list option modifier: %w", err)
 	}
 
+	// If CiliumEndpointSlice feature is enabled, Cilium-agent watches CiliumEndpointSlice
+	// objects instead of CiliumEndpoints. Hence, skip watching CiliumEndpoints if CiliumEndpointSlice
+	// feature is enabled.
+	if option.Config.EnableCiliumEndpointSlice {
+		updatedResources := []string{}
+		for _, r := range resources {
+			if r == k8sAPIGroupCiliumEndpointV2 {
+				continue
+			}
+			updatedResources = append(updatedResources, r)
+		}
+		resources = updatedResources
+	}
+
```

Comment:
The main reason, to  disable watcher for `CEP` in `EnableK8sWatcher` is to make sure we are disabling only `watch events`. `synced` package is also used for registering CRDs with `k8s-apiserver`, disabling in synced package also disable registering the `CEP`s . If CiliumEndpointSlice feature is enabled, we want to register both `CES` and `CEP` and from cilium-agent watch only `CES`.

Reply:
I think there's a similar case in `K8sAPIGroupEndpointV1Core` vs. `K8sAPIGroupEndpointSliceV1Discovery`? For that feature as I undesrtand, we always add the regular resource but conditionally add the Slice version to `resources`. Then in the loop below the Slice version is a noop, and the regular version calls into a function that handles both cases and decides which watcher to register (see `initEndpointsOrSlices()`).

Again, this is bit of a nit, and the above is not a perfect solution, but I'd rather have one approach used consistently across multiple resources rather than using different approaches for different resources like this and embedding this extra loop of CES-specific logic in an otherwise pretty generic loop.

Reply:
Refactored code `CiliumEndpoint` and `CiliumEndpointSlice` similar to `K8sAPIGroupEndpointSliceV1Discovery` pattern. Hopefully `CiliumEndpointSlice` wouldn't look odd implementation for watching `CiliumEndpoint` or `CiliumEndpointSlice`